packages/teams-js/src/private/externalAppCommands.ts

```diff
@@ -0,0 +1,192 @@
+import { sendMessageToParentAsync } from '../internal/communication';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { ApiName, ApiVersionNumber, getApiVersionTag } from '../internal/telemetry';
+import { FrameContexts } from '../public';
+import { errorNotSupportedOnPlatform } from '../public/constants';
+import { runtime } from '../public/runtime';
+
+/**
+ * v2 APIs telemetry file: All of APIs in this capability file should send out API version v2 ONLY
+ */
+const externalAppCommandsTelemetryVersionNumber: ApiVersionNumber = ApiVersionNumber.V_2;
+
+/**
+ * @hidden
+ * Namespace to delegate the ActionCommand to the host
+ * @internal
+ * Limited to Microsoft-internal use
+ *
+ * @beta
+ */
+export namespace externalAppCommands {
+  /**
+   * @hidden
+   * The payload of IActionCommandResponse
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   *
+   * @beta
+   */
+  export interface IActionCommandResponse {
+    /**
+     * type will be result or message
+     */
+    type?: ResultTypes | null;
+    /**
+     * attachment items will exist only when type is result
```

Comment:
>attachment items will exist only when type is result [](http://example.com/codeflow?start=7&length=52)

If this is true, please restructure the interface so that there's no `items` property when the `type` isn't `result`. Consider using a type union or inheritance to achieve this.

Reply:
Same for `text` property but for the opposite `type`  😁

Reply:
Out-dated data model in the latest docs.